### PR TITLE
Bug 1063110: switch to base.validator.

### PR DESCRIPTION
**Pull request by mihneadb**
r? @gregarndt @jonasfj

### CODE REVIEW

**Comment by mihneadb** [comment]
Used `push --force` to clean up the history a bit. Should work now.

File: lib/task.js [hunk 1]
```diff
@@ -599,10 +598,11 @@ export default class Task {
       return await this.abortRun(this.taskState);
     }
     // Validate the schema!
-    let payloadErrors =
-      this.runtime.schema.validate(this.task.payload, PAYLOAD_SCHEMA);
+    let payloadErrors = await validator().then(function (validator) {
+      return validator.check(this.task.payload, payloadSchema);
```
Comment (EggyLv999) [inline lib/task.js]: Try ```js
() => {return validator.check(this.task.payload, payloadSchema);
}```
in place of the anonymous function. As it stands, (I believe) the contextual `this` may be set to the wrong thing.

Comment (gregarndt) [inline lib/task.js]: Ah, it might be.  using arrow functions now makes this a lot less painful as it always captures the context of the surrounding scope.

Comment (mihneadb) [inline lib/task.js]: Forgot about the gotcha, thanks! :+1:

**Comment by mihneadb** [comment]
@gregarndt seems ok now, pls take another look. Thanks!

**Comment by gregarndt** [comment]
This is looking good to me and tests pass.  @jonasfj things look good to you? I haven't used the validator before, but from what I've read it seems sane.

**Comment by mihneadb** [comment]
@gregarndt good to hear, thanks. After this gets merged I'll fix the other bug (default cmd args).

File: bin/worker.js [hunk 2]
```diff
@@ -172,7 +172,6 @@ async function main () {
 
   config.scheduler =
     new taskcluster.Scheduler({ credentials: config.taskcluster });
-  config.schema = require('../lib/schema')();
```
Comment (jonasfj) [inline bin/worker.js]: This should be something like:
```js
config.validator = await base.validator();
config.validator.register(require('../schemas/payload')); // or whereever its...
```
This way we only create the validator object once... And in a future where we have more than one schema we can load them globally...

File: lib/task.js [hunk 3]
```diff
@@ -13,14 +14,12 @@ import getHostname from './util/hostname';
...
 import { pullDockerImage, IMAGE_ERROR } from './pull_image_to_stream';
 import { scopeMatch } from 'taskcluster-base/utils';
+import payloadSchema from '../schemas/payload';
 import waitForEvent from './wait_for_event';
 import _ from 'lodash';
 
 let debug = new Debug('runTask');
 
-const PAYLOAD_SCHEMA =
-  'http://schemas.taskcluster.net/docker-worker/v1/payload.json#';
```
Comment (jonasfj) [inline lib/task.js]: Probably need to keep this. see comment below...

File: lib/task.js [hunk 4]
```diff
@@ -599,10 +598,11 @@ export default class Task {
       return await this.abortRun(this.taskState);
     }
     // Validate the schema!
-    let payloadErrors =
-      this.runtime.schema.validate(this.task.payload, PAYLOAD_SCHEMA);
+    let payloadErrors = await validator().then((validator) => {
```
Comment (jonasfj) [inline lib/task.js]: See change above about global validator instance... then this becomes:
let payloadErrors = this.runtime.validator.check(this.task.payload, PAYLOAD_SCHEMA);

**Comment by jonasfj** [comment]
Using a global instance of `validator` is much smarter... especially if we have multiple schemas... But also necessary in a future where we perhaps have a more expensive compilation step converting JSON schemas into JSON code... the `is-my-json-valid` module does this... and I want to do that in `base.validator` in some future :)

**Comment by mihneadb** [comment]
@jonasfj I've addressed your comments.

**Comment by jonasfj** [comment]
Looks good to me... @gregarndt, are you merging?

**Pull request merged**